packages/http-client-java/generator/http-client-generator-clientcore-test/src/test/java/server/versions/VersionedTests.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package server.versions;
+
+import org.junit.jupiter.api.Test;
+import server.versions.versioned.VersionedClient;
+import server.versions.versioned.VersionedClientBuilder;
+import server.versions.versioned.VersionedServiceVersion;
+
+public class VersionedTests {
+
+    private final VersionedClient client = new VersionedClientBuilder().endpoint("http://localhost:3000")
+        .serviceVersion(VersionedServiceVersion.V2022_12_01_PREVIEW)
+        .buildClient();
+
+    @Test
+    public void test() {
+        client.withoutApiVersion();
+
+        client.withPathApiVersion();
+
+        client.withQueryApiVersion();
+
+        // Request validation failed: Expected query param api-version=2021-01-01-preview but got 2022-12-01-preview
+//        VersionedClient oldClient = new VersionedClientBuilder().endpoint("http://localhost:3000")
+//            .serviceVersion(VersionedServiceVersion.V2021_01_01_PREVIEW)
+//            .buildClient();
+//        oldClient.withQueryOldApiVersion();
```

Comment:
Is this the only test we disabled?

Reply:
I've merged the fix https://github.com/microsoft/typespec/pull/5616

Please refresh main and enable this code.

Reply:
Fixed by [85db78d](https://github.com/microsoft/typespec/pull/5595/commits/85db78dbe9bc06939642db421afd1f2756761c4a)